F23/oct19_ts_fetch/ts_fetch/src/App.tsx

```diff
@@ -0,0 +1,206 @@
+import { useState } from 'react'
+import reactLogo from './assets/react.svg'
+import viteLogo from '/vite.svg'
+import './App.css'
+
+function App() {
+  const [count, setCount] = useState(0)
+
+  return (
+    <>
+      <div>
+        <a href="https://vitejs.dev" target="_blank">
+          <img src={viteLogo} className="logo" alt="Vite logo" />
+        </a>
+        <a href="https://react.dev" target="_blank">
+          <img src={reactLogo} className="logo react" alt="React logo" />
+        </a>
+      </div>
+      <h1>Vite + React</h1>
+      <div className="card">
+        <button onClick={() => setCount((count) => count + 1)}>
+          count is {count}
+        </button>
+        <p>
+          Edit <code>src/App.tsx</code> and save to test HMR
+        </p>
+      </div>
+      <p className="read-the-docs">
+        Click on the Vite and React logos to learn more
+      </p>
+
+      <button onClick={printGridInfo}>Print grid info</button>
+      <button onClick={notThreadsDemo}>Not threads demo</button>
+    </>
+  )
+}
+
+export default App
+
+
+/** 
```

Comment:
Could we move this into its own file; perhaps that would make the code more readable. Right now there is a lot here, and I don't easily see when the part used by React ends.

Reply:
Yeah, I really should do that.